src/components/MainPage.js

```diff
@@ -0,0 +1,111 @@
+import React, { useState, useRef } from 'react';
+import { useSelector } from 'react-redux';
+import { Link } from 'react-router-dom';
+import { FaFacebookF, FaVimeoV } from 'react-icons/fa';
+import { BsTwitter } from 'react-icons/bs';
+import { Swiper, SwiperSlide } from 'swiper/react';
+import 'swiper/css';
+import 'swiper/css/bundle';
+import '../stylesheets/mainpage.css';
+
+const MainPage = () => {
+  const books = useSelector((state) => state.books);
+  const [currentPage] = useState(1);
+  const booksPerPage = 4;
+
+  const startIndex = (currentPage - 1) * booksPerPage;
+  const endIndex = startIndex + booksPerPage;
+  const displayedBooks = books.books.slice(startIndex, endIndex);
+
+  const swiperRef = useRef(null);
+
+  const handlePrevSlide = () => {
+    if (swiperRef.current) {
+      swiperRef.current.slidePrev();
+    }
+  };
```

Comment:
[OPTIONAL]

- The page has some responsiveness issues. We will fix them later on